hedera-node/hedera-admin-service-impl/src/main/java/com/hedera/node/app/service/admin/impl/WritableSpecialFileStore.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.admin.impl;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.spi.state.WritableKVState;
+import com.hedera.node.app.spi.state.WritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Optional;
+
+/**
+ * A {@link WritableSpecialFileStore} is a store for special files that are persisted in the state.
+ *
+ * This class will be moved to network service. It should be deleted here once it is moved over.
+ */
+public class WritableSpecialFileStore {
+    /** The underlying data storage class that holds the file data.
+     * Can contain files 0.0.150 to 159 */
+    private final WritableKVState<Long, byte[]> upgradeFilesById;
```

Comment:
Because the special files can be large (10's of MB's), we will not be able to use any existing `KVState` implementation to store these files by id.

The reason is that hashing such large Merkle leaves using Platform's "built-in" hashing strategy (serialize to `byte[]`, hash the `byte[]`) will be very, very slow.